src/components/LocationRow.jsx

```diff
@@ -0,0 +1,55 @@
+import { urlWithParams } from "../lib/utilityFunctions";
+import { useMeasurements } from "../lib/hooks/swr-extensions";
+import { PreferenceContext } from "../pages/_app";
+import {useMemo, useContext} from "react";
+import { useSensorTypes } from "src/lib/hooks/useSensorTypes";
+import styles from "src/styles/LocationRow.module.css"
+import { capitalize } from "src/lib/utilityFunctions";
+export default function LocationRow({locName, selected}){
+    
+    const ENDPOINT = "/api/v3/measurements?";
+
+    const { preferences } = useContext(PreferenceContext);
+    const sensorTypes = useSensorTypes("/api/v3/sensors/types");
+
+    const url = useMemo(() => urlWithParams(ENDPOINT, {
+        pageSize: 1,
+        sortOrder: "desc",
+        locationName: locName,
+        temperatureUnit: preferences.temperatureUnit.symbol,
+        conductivityUnit: preferences.conductivityUnit.symbol,
+    }), [preferences]);
+    
+    const { measurements, pagination, isLoading, isLagging, error } = useMeasurements(url, {refreshInterval: 5000});
+    
+    if (error) {
+        return <div>No data</div>;
+    }
+    
+    if(isLoading || !sensorTypes)
+    return <div>Loading again...</div>;

```

Comment:
Usage of [ProgressBar](https://github.com/sensor-network/open-data-portal/blob/integrations/src/components/CustomProgressBar.tsx)?